transactions/serializers.py

```diff
@@ -32,3 +40,122 @@ class TransactionModelSerializer(serializers.ModelSerializer):
+
+        try:
+            assert validated_data["tx_inputs_count"] == len(inputs)
+            assert validated_data["tx_outputs_count"] == len(outputs)
+
+        except AssertionError:
+            raise ValidationError("Number of inputs and outputs specified should match the actual number of inputs and outputs")
+
+        '''
+        Check if sum of inputs is greater than equal to sum of outputs
+        '''
+        burnt = reduce(lambda x, y: x + y, map(lambda i: i["value"], inputs), 0)
+        minted = reduce(lambda x, y: x + y, map(lambda o: o["value"], outputs), 0)
+
+        if minted > burnt:
+            raise ValidationError("Sum of inputs can not be less than sum of outputs")
+
+        '''
+        Check if all reference UTXOs are indeed valid, calculate and store hashes
+        '''
+
+        input_hashes = list(map(lambda i: self.validate_input(i), inputs))
+        output_hashes = list(map(lambda o: hashlib.sha1((str(o["value"]) + o["own_addr"]).encode()).hexdigest(), outputs))
+
+        for idx in range(len(inputs)):
+            inputs[idx]['hash'] = input_hashes[idx]
+
+        for idx in range(len(outputs)):
+            outputs[idx]['hash'] = output_hashes[idx]
+
+        '''
+        Check signatures
+        '''
+
+        tx_hash_content = str(validated_data["tx_inputs_count"]) + reduce(lambda a,b: a+b, input_hashes, "") + \
+                          str(validated_data["tx_outputs_count"]) + reduce(lambda a,b: a+b, output_hashes, "")
+        
+        tx_hash = hashlib.sha1(tx_hash_content.encode()).hexdigest()
+
+        sign_statuses = map(lambda i: self.sign_status(tx_hash, i["own_addr"], i["signature"]), inputs)
+
+        if not all(sign_statuses):
+            raise ValidationError("One or more Signatures Do not Match")
+        
+        validated_data["tx_hash"] = tx_hash
+
+        return validated_data
+
+    def create(self, validated_data):
+        inputs = validated_data['inputs']
+        outputs = validated_data['outputs']
+
+        '''
+        Create the posted transaction. Update the database accordingly
+        '''
+
+        t = Transaction.objects.create(hash=validated_data['tx_hash'], tx_inputs_count=validated_data['tx_inputs_count'], tx_outputs_count=validated_data['tx_outputs_count'])
+
+        for idx in range(len(inputs)):
+
+            i = inputs[idx]
+            gen_tx = Transaction.objects.filter(hash=i['gen_transaction']['hash']).first()
+            TransactionInput.objects.create(value=i['value'], own_addr=i['own_addr'], gen_transaction=gen_tx, gen_transaction_index=i['gen_transaction_index'], spend_transaction_index=idx, spend_transaction=t, signature=i['signature'], hash=i['hash'])
+
+            '''
+            Now this is a used up utxo. So delete it
+            '''
+            txouts_to_delete = TransactionOutput.objects.filter(value=i['value'], own_addr=i['own_addr'], gen_transaction=gen_tx, gen_transaction_index=i['gen_transaction_index'])
```

Comment:
can we somehow disable (or set status) it instead of delete?

Reply:
we delete it from the outputs and add the information contained(in it) in the form of an input...see the line just above where we store a transactionInput consisting of roughly the same information